cs/tools.md

```diff
@@ -0,0 +1,393 @@
+- Zdarma.
+
+Nevýhody:
+- Nutné do instalovat pluginy pro vývoj v php a laravelu.
+- Náročnější refactoring oproti PHP Strom.
+
+https://code.visualstudio.com/
+
+
+<a name="sublime-text"></a>
+
+### Sublime Text
+
+Výborný nástroj na práci s velkými soubory jako jsou dumpy z databáze, XML nebo json soubory.
+
+Výhody:
+- Zvládá otevírání a práci s velkými soubory.
+- Zdarma.
+
+Nevýhody:
+- Podpora pluginů není udržována jako ve VS Code nebo PHP storm.
+- Občasné chyby pluginů.
+
+https://www.sublimetext.com/
+
+
+## Debug / Testování
+
+
+<a name="spatie-ray"></a>
+
+### Spatie Ray
+
+Nástroj pro debug php bez zásahu do flow aplikace.
+Už žádné echo, dd(), dump() (Pouze sranda pořád jsou případy kdy je efektivnější napsat dd).
+Ray Vám dá možnost analyzovat aplikaci efektivně s možností rozdělení debugovaných částí kódu do jednotlivých záložek dle barev.
+
+Výhody:
+- Efektivní debug aplikace.
+- Snadná organizace dumpů dle barev.
+- Přehledný rozpis zápisů na jednom místě.
+- Hodně variant výstupů debubovaného kódu.
+
+Nevýhody:
+- Při velkém množství zápisů je aplikace pomalejší.
+- Placená licence (Úhel pohledu).
+
+https://spatie.be/products/ray
+
+
+<a name="tinkerwell"></a>
+
+### Tinkerwell
+
+Nástroj pro snadné testování kódu přímo na projektu s možností připojení přes ssh na remote server a spouštění scriptů na něm.
+
+Výhody:
+- Efektivní testování logiky kódu na daném projektu.
+- Spouštění kódu v projektu bez použití browseru.
+- Možnost ukládání scriptů do snippets.
+- Pouštění scriptů přes ssh přímo na serverech. Například pro update dat v databázi.
+
+Nevýhody:
+- Placená licence (Úhel pohledu).
+
+https://tinkerwell.app/
+
+<a name="postman"></a>
+
+### Postman
+
+Nástroj pro testování api.
+
+https://www.postman.com/
+
+
+<a name="insomnia"></a>
+
+### Insomnia
+
+Nástroj na design, debug a testování API.
+
+https://insomnia.rest/
+
+
+## Databáze
+
+
+<a name="table-plus"></a>
+
+### Table Plus
+
+Úžasný nástroj pro správu nejrůznějších databází jak na lokálním environmentu tak na serverech.
+
+Výhody:
+- Správa databází na jednom místě.
+- SSH připojení na servery.
+- Přehledné UI.
+- Spouštění raw queries přímo na lokálním projektu nebo na serveru.
+- Debug queries.
+
+Nevýhody:
+- Free licence dovolí otevřít pouze 3 záložky.
+
+https://tinkerwell.app/
+
+<a name="dbngin"></a>
+
+### DBngin
+
+Rychlá instalace a management lokálních databází jako je PostgreSQL, MySQL, Redis.
+
+https://dbngin.com/
+
+
+## Emaily
+
+<a name="mailtrap"></a>
+
+### MailTrap
+
+Cloudová služba pro testování emailů jak na lokálním environmentu tak na serveru.
+
+https://mailtrap.io/
+
+<a name="helo"></a>
+
+### Helo
+
+Nástroj pro testování emailů na lokálním environmentu.
+
+https://usehelo.com/
+
+## Lokální vývoj
+
+<a name="valet"></a>
+
+### Valet
+
+Snadný způsob jak spravovat projekty a lokální server. Díky valetu můžete snadno spouštět nové projekty a okamžitě na nich pracovat bez nutnosti nastavování lokálního prostředí.
+
```

Comment:
Lze snadno měnit PHP verzi pro jednotlivé projekty přes isolate.

Reply:
To uz sa viac aktualne hodi https://herd.laravel.com ako PHP Monitor